app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -9,108 +9,78 @@ module IdentityProviders
           fetch_identity_path: Authentication::AuthnJwt::IdentityProviders::FetchIdentityPath.new,
           fetch_authenticator_secrets: Authentication::Util::FetchAuthenticatorSecrets.new,
           check_authenticator_secret_exists: Authentication::Util::CheckAuthenticatorSecretExists.new,
-          add_prefix_to_identity: Authentication::AuthnJwt::IdentityProviders::AddPrefixToIdentity.new,
           logger: Rails.logger
         },
         inputs: %i[jwt_authenticator_input]
       ) do
-        extend(Forwardable)
-        def_delegators(:@jwt_authenticator_input, :service_id, :authenticator_name, :account)
 
         def call
           @logger.debug(
             LogMessages::Authentication::AuthnJwt::FetchingIdentityByInterface.new(
               TOKEN_IDENTITY_PROVIDER_INTERFACE_NAME
             )
           )
-          fetch_identity_name_from_token
-          fetch_identity_path
-          add_prefix_path_to_identity_name
-          add_host_to_identity_with_prefix
+
+          # Ensures token has id claim, and stores its value in @id_from_token.
+          fetch_id_from_token
+
+          # Get value of "identity-path", which is stored as a Conjur secret.
+          id_path = @fetch_identity_path.call(
+            jwt_authenticator_input: @jwt_authenticator_input
+          )
+
+          # Create final id by joining "host", <path>, and <id>.
+          host_prefix = IDENTITY_TYPE_HOST
+          full_host_id = AuthnJwt.path_join(host_prefix, id_path, @id_from_token)
+
           @logger.info(
             LogMessages::Authentication::AuthnJwt::FetchedIdentityByInterface.new(
-              host_identity_with_prefix,
+              full_host_id,
               TOKEN_IDENTITY_PROVIDER_INTERFACE_NAME
             )
           )
 
-          host_identity_with_prefix
+          full_host_id
         end
 
         private
 
-        def fetch_identity_name_from_token
-          return @identity_name_from_token if @identity_name_from_token
+        def fetch_id_from_token
+          return @id_from_token if @id_from_token
+
+          @logger.debug(
+            LogMessages::Authentication::AuthnJwt::CheckingIdentityFieldExists.new(id_claim_key)
+          )
+
+          @id_from_token = @jwt_authenticator_input.decoded_token[id_claim_key]
 
-          @logger.debug(LogMessages::Authentication::AuthnJwt::CheckingIdentityFieldExists.new(token_id_field_secret))
-          @identity_name_from_token = decoded_token[token_id_field_secret]
-          if @identity_name_from_token.blank?
-            raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken, token_id_field_secret
+          # Equivalent to Rails blank? but without a Rails dependency.
```

Comment:
Conjur is Rails application what the problem depanding on rails ?

Reply:
@tzheleznyak Nothing under `app/domain` should have any dependency on Rails.  Indeed, the whole point of that directory was to decouple from Rails.

From [`app/domain` README](https://github.com/cyberark/conjur/blob/master/app/domain/README.md#introduction):

> You can think of app/domain as a haven where pure Conjur business logic (the Conjur "domain") lives as plain ruby code, unpolluted by framework-specific concepts like controllers or routes. The code in this folder should have no dependencies on Rails itself (if you find any, they are likely mistakes we should fix).

One of the major downsides of Rails is the way that code from anywhere can implicitly access code from anywhere else -- creating a network of hidden dependencies.  You can't know what some piece of code depends on without reading through all of the code and manually checking all the references.   In the `app/domain` code, we want all the dependencies to be explicit, and so we have to opt out of the "Rails soup". 

We've even talked about pulling the authenticators out into their own gems, to _enforce_ this separation.  Which would be great.  But doing that adds complexity in other ways, so we haven't done yet.  So until we have a better solution, we just have to rely on developers knowing that this is a rule.